### PR TITLE
Emily openapi template generation - (OpenAPI Client 1/3)

### PR DESCRIPTION
**Pull request by AshtonStephens**
## Description

Closes: #283 

## Changes

Changes type names an imports in order to make it possible to autogenerate the openapi specification.

I had to remove all type aliases in order for this PR to play nicely with the openapi autogeneration.

## Checklist:

- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: emily/handler/src/api/handlers/chainstate.rs [hunk 1]
```diff
@@ -41,10 +34,7 @@ pub async fn get_chain_tip(context: EmilyContext) -> impl warp::reply::Reply {
         // TODO(390): Handle multiple being in the tip list here.
         let api_state = accessors::get_api_state(&context).await?;
         let chaintip: Chainstate = api_state.chaintip().into();
-        Ok(with_status(
-            json(&(chaintip as GetChainstateResponse)),
-            StatusCode::OK,
-        ))
+        Ok(with_status(json(&(chaintip as Chainstate)), StatusCode::OK))
```
Comment (djordon) [inline emily/handler/src/api/handlers/chainstate.rs:37]: The `as` cast seems unnecessary, given the line above. Same for a few of these functions.

File: emily/handler/src/api/handlers/chainstate.rs [hunk 2]
```diff
@@ -101,31 +88,28 @@ pub async fn get_chainstate_at_height(
...
-        body: SetChainstateRequestBody,
+        body: Chainstate,
     ) -> Result<impl warp::reply::Reply, Error> {
         // Convert body to the correct type.
         let chainstate: Chainstate = body;
         add_chainstate_entry_or_reorg(&context, &chainstate).await?;
         // Respond.
         Ok(with_status(
-            json(&(chainstate as SetChainstateResponse)),
+            json(&(chainstate as Chainstate)),
```
Comment (djordon) [inline emily/handler/src/api/handlers/chainstate.rs:112]: The move on line 108 seems unnecessary, couldn't you just rename the `body` input variable to `chainstate` or just use body throughout?

File: emily/handler/src/api/models/withdrawal/mod.rs [hunk 3]
```diff
@@ -99,23 +96,23 @@ pub struct WithdrawalParameters {
...
+    pub stacks_block_hash: String,
     /// The height of the Stacks block in which this request id was initiated.
-    pub stacks_block_height: BlockHeight,
+    pub stacks_block_height: u64,
     /// The recipient Bitcoin address.
-    pub recipient: BitcoinAddress,
+    pub recipient: String,
     /// Amount of BTC being withdrawn.
-    pub amount: Satoshis,
+    pub amount: u64,
```
Comment (djordon) [inline emily/handler/src/api/models/withdrawal/mod.rs:107]: I would update the comment to state that this is in sats, although maybe that is obvious given that the type is `u64`?

**Review by djordon — COMMENTED** [review]
Looks good, I'd just remove some of the casts in the handler code.

File: emily/handler/src/api/handlers/chainstate.rs (hunk 1 above)
Comment (AshtonStephens) [reply emily/handler/src/api/handlers/chainstate.rs:37]: I'll fix that. Yeah it's unnecessary

File: emily/handler/src/api/models/withdrawal/mod.rs (hunk 3 above)
Comment (AshtonStephens) [reply emily/handler/src/api/models/withdrawal/mod.rs:107]: I agree it's not obvious. I'll fix it

Comment (AshtonStephens) [reply emily/handler/src/api/models/withdrawal/mod.rs:107]: Changed the comment above.

```rust
    /// Amount of BTC being withdrawn in satoshis.
    pub amount: u64,
```

File: emily/handler/src/database/entries/deposit.rs [hunk 4]
```diff
@@ -251,11 +248,11 @@ impl TryFrom<DepositEntry> for Deposit {
 pub struct DepositParametersEntry {
     /// Maximum fee the signers are allowed to take from the deposit to facilitate
     /// the transaction.
-    pub max_fee: Satoshis,
+    pub max_fee: u64,
```
Comment (djordon) [inline emily/handler/src/database/entries/deposit.rs:251]: I think we should note the denomination for the `max_fee` fields as well.

**Review by djordon — APPROVED** [review]
We should probably update `max_fee` fields too. Oh and we probably need to do `cargo fmt`. But besides that it looks good!

**Pull request merged**